aten/src/ATen/core/type.cpp

```diff
@@ -1166,19 +1182,21 @@ size_t ClassType::addAttribute(
         "Expecting parameter to have either None, Tensor or Optional[Tensor] type, but got: ",
         toString(type));
   }
-  if (is_module()) {
-    parameterSlots_->push_back(is_parameter);
+  if (was_registered_as_buffer) {
+      TORCH_INTERNAL_ASSERT(is_module(), "attempting to register a buffer on a non module");
+    TORCH_CHECK((type->kind() == TensorType::Kind),
```

Comment:
This check should be the same as the above. The PT API allows users to register `None`, `Tensor`, or `Optional[Tensor]` as a buffer.